gobblin-runtime/src/main/java/org/apache/gobblin/runtime/api/FlowSpec.java

```diff
@@ -125,6 +128,19 @@ public static FlowSpec.Builder builder(URI catalogURI, Properties flowProps) {
     }
   }
 
+  /**
+   * Add property to Config (also propagated to the Properties field)
+   * @param key
+   * @param value
+   */
+  public void addProperty(String key, String value) {
```

Comment:
since this modifies two fields, instances now require synchronization around all access of either field

Reply:
I added the `volatile` keyword and note about accessing both parameters with synchronized key word if multi-threaded access (which is unlikely since we'd probably only access one not both). When trying to use synchronized keyword on the two functions it results in a number of `Inconsistent synchronization of org.apache.gobblin.runtime.api.FlowSpec.config; locked 50% of time` errors so I had to remove the `@Data` annotation for the class and would have to add separate `@Getter/Setter/toString` etc... annotations for all the fields to make only `Config/ConfigAsProperties` synchronized. I don't think this is that useful so made the compromise above. Let me know.

Reply:
Probably overkill but I think setting a rw lock would handle this right?

Reply:
This will also remove the findBugsMain, because by annotating them `volatile` we are telling it that these fields can be changed from other threads.

Reply:
I updated the fields to be private so they are only updated through the addProperty method and bc we don't expect this scenario to occur I don't want to over complicate with rw lock for now. Future developers should be able to quickly notice this with the volatile keyword and documentation.